### PR TITLE
[FUTURE] cmake: Switch from tri-state options to ON/OFF

### PR DESCRIPTION
**Pull request by hebasto**
This PR is a continuation of the today's (2024-01-25) discussion on the meeting.

It is supposed to be a place for the further discussion and making a decision regarding the default option values.

---

The `cmake-adjusted` branch is to be always synced with the `cmake-staging` one.

### CODE REVIEW

File: .github/workflows/cmake.yml [hunk 1]
```diff
@@ -113,29 +113,29 @@ jobs:
...
             depends_options: 'NO_SQLITE=1 NO_USDT=1'
-            build_options: '-DWITH_BDB=ON'
+            build_options: '-DENABLE_WALLET=ON -DWITH_BDB=ON'
             expected: 'pass'
 
           # SQLite: Using CMake's FindSQLite3 module.
           - system_packages: 'libboost-dev libevent-dev'
             depends_options: 'NO_BDB=1 NO_SQLITE=1 NO_USDT=1'
-            build_options: '-DWITH_SQLITE=ON'
+            build_options: '-DENABLE_WALLET=ON -DWITH_SQLITE=ON'
```
Comment (fanquake) [inline .github/workflows/cmake.yml:130]: Shouldn't `WITH_SQLITE=ON` imply `ENABLE_WALLET`? What (extra) does `ENABLE_WALLET` do here? I'm not sure we'll actually want to keep it as an option.

**Comment by hebasto** [comment]
See https://github.com/hebasto/bitcoin/pull/161 as a stage one of switching away from tri-state options.

File: .github/workflows/cmake.yml (hunk 1 above)
Comment (hebasto) [reply .github/workflows/cmake.yml:130]: Reworked in https://github.com/hebasto/bitcoin/pull/164.

**Comment by hebasto** [comment]
> See #161 as a stage one of switching away from tri-state options.

Closing this PR. Please review the following ones:
- https://github.com/hebasto/bitcoin/pull/162
- https://github.com/hebasto/bitcoin/pull/164
- https://github.com/hebasto/bitcoin/pull/169
- https://github.com/hebasto/bitcoin/pull/173

**Pull request closed without merging**